src/index.ts

```diff
@@ -115,21 +115,85 @@ function makePassesFilter(opts: FilterOpts): (msg: Msg) => boolean {
   return () => true;
 }
 
+function assertFollowingOnlyUsability<
+  T extends ((obj: any, cb: CB<boolean>) => void) | null,
+>(enabled: boolean, isFollowing: T): asserts isFollowing is NonNullable<T> {
```

Comment:
Cool, I did not know about the existence of `asserts ... is NonNullable<T>` in TypeScript.

Reply:
yeah basically an extension of the `is` keyword and is useful for cases where the asserted variable is used way later in the scope of the calling function. I've used it sparingly but handy to know of